From 56b8ae85823953efd27bed0ad57ca669875712dc Mon Sep 17 00:00:00 2001
From: Jim Somerville <Jim.Somerville@windriver.com>
Date: Mon, 14 Apr 2014 17:54:32 -0400
Subject: [PATCH 3/5] Fix vcpuaffine feature test mask

Also disable the feature by default

Signed-off-by: Jim Somerville <Jim.Somerville@windriver.com>
[ywei: fix context conflict in
	hw/i386/kvm/pci-assign.c]
Signed-off-by: Yunguo Wei <yunguo.wei@windriver.com>
Upstream-Status: Pending
Signed-off-by: He Zhe <zhe.he@windriver.com>
---
 hw/i386/kvm/pci-assign.c | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/hw/i386/kvm/pci-assign.c b/hw/i386/kvm/pci-assign.c
index 14ea8827..599dc5ad 100644
--- a/hw/i386/kvm/pci-assign.c
+++ b/hw/i386/kvm/pci-assign.c
@@ -83,6 +83,7 @@ typedef struct AssignedDevRegion {
 
 #define ASSIGNED_DEVICE_PREFER_MSI_MASK (1 << ASSIGNED_DEVICE_PREFER_MSI_BIT)
 #define ASSIGNED_DEVICE_SHARE_INTX_MASK (1 << ASSIGNED_DEVICE_SHARE_INTX_BIT)
+#define ASSIGNED_DEVICE_VCPU_AFFINITY_MASK (1 << ASSIGNED_DEVICE_VCPU_AFFINITY_BIT)
 
 typedef struct MSIXTableEntry {
     uint32_t addr_lo;
@@ -1570,7 +1571,7 @@ static void kvm_msix_update_affinity(AssignedDevice *dev, int entry, int dest)
     int irq, fd, i;
     cpu_set_t *msk;
 
-    if (!(dev->features & ASSIGNED_DEVICE_VCPU_AFFINITY_BIT))
+    if (!(dev->features & ASSIGNED_DEVICE_VCPU_AFFINITY_MASK))
         return;
 
     irq = assign_query_msix_irq(dev, entry);
@@ -1922,7 +1923,7 @@ static Property assigned_dev_properties[] = {
     DEFINE_PROP_BIT("share_intx", AssignedDevice, features,
                     ASSIGNED_DEVICE_SHARE_INTX_BIT, true),
     DEFINE_PROP_BIT("vcpuaffine", AssignedDevice, features,
-                        ASSIGNED_DEVICE_VCPU_AFFINITY_BIT, true),
+                        ASSIGNED_DEVICE_VCPU_AFFINITY_BIT, false),
     DEFINE_PROP_STRING("configfd", AssignedDevice, configfd_name),
     DEFINE_PROP_END_OF_LIST(),
 };
-- 
2.13.3

